File: Dataset/OOP_prepared/Java/Resizable_trace_4.java
public interface GeometricObject {
    public double getArea();
    public double getPerimeter();
}

public class Circle implements GeometricObject {
    protected double radius;

    public Circle(double radius) {
        this.radius = radius;
    }

    public Circle() {
    }

    public double getRadius() {
        return radius;
    }

    public void setRadius(double radius) {
        this.radius = radius;
    }

    @Override
    public String toString() {
        return "Circle[" + "radius=" + radius + ']';
    }

    @Override
    public double getArea() {
        return radius * radius * Math.PI;
    }

    @Override
    public double getPerimeter() {
        return radius * 2 * Math.PI;
    }
}

public interface Resizable {
    public abstract void resize(int percent);
}

public class ResizableCircle extends Circle implements Resizable {
    public ResizableCircle(double radius) {
        super(radius);
    }

    public ResizableCircle() {
    }

    @Override
    public String toString() {
        return "ResizableCircle[" + super.toString() + ']';
    }

    @Override
    public void resize(int percent) {
        radius *= percent / 100.0;
    }
}

public class TestMain {
    public static void main(String[] args) {
        GeometricObject g1 = new Circle(8.1);
        System.out.println(g1);
        System.out.println("Perimeter = " + g1.getPerimeter());
        System.out.println("Area = " + g1.getArea());
    }
}

File: Dataset/Advanced/Test/OOP_prepared/Python/Resizable_trace_4.py
import math
from abc import ABC, abstractmethod

class GeometricObject(ABC):
    @abstractmethod
    def get_area(self):
        pass

    @abstractmethod
    def get_perimeter(self):
        pass

class Circle(GeometricObject):
    def __init__(self, radius=0.0):
        self._radius = radius

    @property
    def radius(self):
        return self._radius

    @radius.setter
    def radius(self, radius):
        self._radius = radius

    def __str__(self):
        return f"Circle[radius={self._radius}]"

    def get_area(self):
        return self._radius * self._radius * math.pi

    def get_perimeter(self):
        return self._radius * 2 * math.pi

class Resizable(ABC):
    @abstractmethod
    def resize(self, percent):
        pass

class ResizableCircle(Circle, Resizable):
    def __init__(self, radius=0.0):
        super().__init__(radius)

    def __str__(self):
        return f"ResizableCircle[{super().__str__()}]"

    def resize(self, percent):
        self._radius *= percent / 100

if __name__ == "__main__":
    g5 = ResizableCircle(6.0)
    g5.resize(50)
    print(g5)
    print(f"New area after resizing: {g5.get_area()}")
    print(f"New perimeter after resizing: {g5.get_perimeter()}")